website/project/views/drafts.py

```diff
@@ -0,0 +1,269 @@
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def edit_draft_registration_page(auth, node, draft, **kwargs):
+    """Draft registration editor
+
+    :return: serialized DraftRegistration
+    :rtype: dict
+    """
+    if draft.registered_node:
+        raise HTTPError(http.FORBIDDEN, data={
+            'message_short': 'This draft has already been registered.',
+            'message_long': 'This draft has already been registered and cannot be modified.'
+        })
+    ret = project_utils.serialize_node(node, auth, primary=True)
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def update_draft_registration(auth, node, draft, *args, **kwargs):
+    """Update an existing draft registration
+
+    :return: serialized draft registration
+    :rtype: dict
+    :raises: HTTPError
+    """
+    if draft.registered_node:
+        raise HTTPError(http.FORBIDDEN, data={
+            'message_short': 'This draft has already been registered.',
+            'message_long': 'This draft has already been registered and cannot be modified.'
+        })
+    data = request.get_json()
+
+    schema_data = data.get('schema_data', {})
+    schema_data = rapply(schema_data, strip_html)
+
+    schema_name = data.get('schema_name')
+    schema_version = data.get('schema_version', 1)
+    if schema_name:
+        meta_schema = get_schema_or_fail(
+            Q('name', 'eq', schema_name) &
+            Q('schema_version', 'eq', schema_version)
+        )
+        existing_schema = draft.registration_schema
+        if (existing_schema.name, existing_schema.schema_version) != (meta_schema.name, meta_schema.schema_version):
+            draft.registration_schema = meta_schema
+
+    draft.update_metadata(schema_data)
+    draft.save()
+    return serialize_draft_registration(draft, auth), http.OK
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def delete_draft_registration(auth, node, draft, *args, **kwargs):
+    """Permanently delete a draft registration
+
+    :return: None
+    :rtype: NoneType
+    """
+    DraftRegistration.remove_one(draft)
+    return None, http.NO_CONTENT
+
+def get_metaschemas(*args, **kwargs):
+    """
+    List metaschemas with which a draft registration may be created. Only fetch the newest version for each schema.
+
+    :return: serialized metaschemas
+    :rtype: dict
+    """
+    count = request.args.get('count', 100)
+    include = request.args.get('include', 'latest')
+
+    meta_schema_collection = database['metaschema']
+
+    meta_schemas = []
+    if include == 'latest':
+        schema_names = meta_schema_collection.distinct('name')
+        for name in schema_names:
+            meta_schema_set = MetaSchema.find(
+                Q('name', 'eq', name) &
+                Q('schema_version', 'eq', 2)
```

Comment:
As long as the UI requires a certain format of schema, I think we need to enforce a rigid version number. Maybe it makes sense to move this to a `website.settings` constant?